src/applications/disability-benefits/all-claims/components/AutoComplete.jsx

```diff
@@ -0,0 +1,177 @@
+
+  const inputRef = useRef(null);
+  const resultsRef = useRef([]);
+
+  // Delays screen reader result count reading to avoid interruption by input content reading
+  const debouncedSetAriaLiveText = useRef(
+    debounce((resultCount, freeTextResult) => {
+      const makePlural = resultCount > 1 ? 's' : '';
+
+      setAriaLiveText(
+        `${resultCount} result${makePlural}. ${freeTextResult}, (1 of ${resultCount})`,
+      );
+    }, 700),
+  ).current;
+
+  const debouncedSearch = useRef(
+    debounce(async inputValue => {
+      const freeTextResult = createFreeTextItem(inputValue);
+      const searchResults = fullStringSimilaritySearch(
+        inputValue,
+        availableResults,
+      );
+      const updatedResults = [freeTextResult, ...searchResults];
+      setResults(updatedResults);
+      setActiveIndex(0);
+
+      debouncedSetAriaLiveText(updatedResults.length, freeTextResult);
+    }, debounceDelay),
+  ).current;
+
+  const closeList = () => {
+    debouncedSearch.cancel();
+    debouncedSetAriaLiveText.cancel();
+    setResults([]);
+    setActiveIndex(null);
+  };
+
+  const handleInputChange = inputValue => {
+    setValue(inputValue);
+    onChange(inputValue);
+
+    if (!inputValue) {
+      closeList();
+      setAriaLiveText('Input is empty. Please enter a condition.');
+      return;
+    }
+
+    debouncedSearch(inputValue);
+  };
+
+  const scrollIntoView = index => {
+    const activeResult = resultsRef.current[index];
+    activeResult?.scrollIntoView({
+      block: 'nearest',
+    });
+  };
+
+  const navigateList = (e, adjustment) => {
+    e.preventDefault();
+    const newIndex = activeIndex + adjustment;
+
+    if (newIndex < 0 || newIndex > results.length - 1) {
+      return;
+    }
+
+    setActiveIndex(newIndex);
+    setAriaLiveText(
+      `${results[newIndex]}, (${newIndex + 1} of ${results.length})`,
+    );
+    scrollIntoView(newIndex);
+  };
+
+  const selectResult = result => {
+    const newValue = result === createFreeTextItem(value) ? value : result;
+    setValue(newValue);
+    onChange(newValue);
+    setAriaLiveText(`${newValue} is selected`);
+    closeList();
+
+    inputRef.current.shadowRoot.querySelector('input').focus();
+  };
+
+  const handleKeyDown = e => {
+    if (results.length > 0) {
+      if (e.key === 'ArrowDown') {
+        navigateList(e, 1);
+      } else if (e.key === 'ArrowUp') {
+        navigateList(e, -1);
+      } else if (e.key === 'Enter') {
+        selectResult(results[activeIndex]);
+      } else if (e.key === 'Escape') {
+        closeList();
+      }
+    }
+  };
+
+  return (
+    <div className="cc-autocomplete">
+      <VaTextInput
+        data-testid="autocomplete-input"
+        id={id}
+        label={label}
+        message-aria-describedby={!value ? INSTRUCTIONS : null}
+        ref={inputRef}
+        required
+        value={value}
+        onBlur={() => setTimeout(closeList, 200)} // Enables clicking option from list
```

Comment:
Best practice is usually to pull inline functions out since they create a new function instance on every render. I don't think we'd really get that much performance benefit, and it does create more functions that aren't reused, so up to you.